src/common/path/platform/c++17/path.cpp

```diff
@@ -112,6 +116,40 @@ expected::ExpectedBool IsExecutable(const string &file_path, const bool warn) {
 	}
 }
 
+expected::ExpectedBool AreFilesIdentical(const string &file_one, const string &file_two) {
```

Comment:
After closer inspection, I realized that this function does not use any of the `c++17` features (`fs` namespace`) so it could very well be made generic and moved to `src/common/path/path.cpp`, then adding it to the same library with:
```
diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index ada11934..b361e43d 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -71,6 +71,7 @@ target_link_libraries(common_io PUBLIC common)
 
 add_library(common_path STATIC)
 target_sources(common_path PRIVATE
+  path/path.cpp
   path/platform/c++17/path.cpp
   path/platform/posix/path.cpp
 )
```